### PR TITLE
Fixed intl parsing bug when trying to parse non-string entries

### PR DESCRIPTION
**Pull request by sepehr500**
Fixed intl parsing bug when trying to parse non-string entries.

---
This change is [Reviewable](https://reviewable.io/reviews/shakacode/react_on_rails/1153)

### CODE REVIEW

**Comment by justin808** [comment]
@sepehr500 please consult with @vcarel regarding #1129.

After that is done, I can assess this.

File: lib/react_on_rails/locales_to_js.rb [hunk 1]
```diff
@@ -107,7 +107,7 @@ def flatten(translations)
       translations.each_with_object({}) do |(k, v), h|
         if v.is_a? Hash
           flatten(v).map { |hk, hv| h["#{k}.#{hk}".to_sym] = hv }
-        elsif !v.is_a? Array # Arrays are not supported by react-intl
+        elsif v.is_a? String # Arrays are not supported by react-intl
```
Comment (vcarel) [inline lib/react_on_rails/locales_to_js.rb]: The comment does not seem appropriate anymore

Comment (sepehr500) [reply lib/react_on_rails/locales_to_js.rb]: @vcarel Do you think we should support int and bools as well as strings?

**Comment by vcarel** [comment]
@sepehr500 What issue does it address? I confess I'm concerned about not exporting empty keys, because react-intl would then output the name of the missing keys instead of just blank strings.

@justin808 The root cause of those issues is that we attempt to pass translations to react-intl which will never be used on React's end. Instead, shouldn't we encouraging users to cleanup their translations files and pass only what React will really use?

File: lib/react_on_rails/locales_to_js.rb (hunk 1 above)
Comment (vcarel) [reply lib/react_on_rails/locales_to_js.rb]: Looks like int and bools are not supported -> https://formatjs.io/guides/message-syntax/

Comment (vcarel) [reply lib/react_on_rails/locales_to_js.rb]: ...I mean by react-intl.
I don't know, see my comment to @justin808 below. I don't feel this is going in the right direction, although my last PR is part of the problem.
The decision is up to both of you.

**Comment by sepehr500** [comment]
@vcarel There are people who use part of their yml for react and the other part for rails. So forcing people to clean up their yml will not work because the libraries support different data types. This PR is to stop the build from failing when you have a type that can not have `.gsub` called on it.

**Comment by justin808** [comment]
Nice job @sepehr500. @vcarel please give this a try asap. I'm pushing 11.0.7 ASAP.

**Pull request merged**